### PR TITLE
I've made a big change and I need a code review before merging to the…

### PR DESCRIPTION
**Pull request by bennevis**
code review is needed for this new wonderful feature

### CODE REVIEW

File: pretend_file1.java [hunk 1]
```diff
@@ -3,3 +3,11 @@
 java code original line 3 Edited DWS on desktop GBPC2185
 jave code original line 4 Edited MF on desktop GBPC2141
 jave code original line 5
+
+/**
+ * 
+ **/
+public interface pretend_fileq1 {
```
Comment (bennevis) [inline pretend_file1.java]: are happy to set this an interface so we can implement this in different ways, for example a recursive or an iterative way

Comment (Falski) [inline pretend_file1.java]: It's indeed a very good idea!

File: pretend_file1.java [hunk 2]
```diff
@@ -3,3 +3,11 @@
...
+  
+  double factorial(int n);
```
Comment (Falski) [inline pretend_file1.java]: @bennevis  'n' is not very intuitive name, could you propose something more readable?

Comment (bennevis) [inline pretend_file1.java]: @Falski , 'number' is ok? :1234:

Comment (Falski) [inline pretend_file1.java]: yes, that'll do :ok_hand: 
thank you :bow:

**Pull request merged**